engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -320,8 +309,11 @@ private void preRenderUpdate(RenderingStage renderingStage) {
     public void render(RenderingStage renderingStage) {
         preRenderUpdate(renderingStage);
 
-        renderShadowMap();          // into shadowMap buffer
-        renderWorldReflection();    // into sceneReflect buffer
+        if (renderingConfig.isDynamicShadows() && isFirstRenderingStageForCurrentFrame) {
+            shadowMapNode.process(); // into shadowMap buffer
+        }
+
+        worldReflectionNode.process(); // into sceneReflect buffer
```

Comment:
In this section, let's move the IF block INTO the shadowMapNode.process() method. Also, rather than processing each node explicitly, you can now iterate the list.